Orange/widgets/data/owconcatenate.py

```diff
@@ -172,18 +179,17 @@ def incompatible_types(self):
         return False
 
     def apply(self):
+        self.Warning.renamed_variables.clear()
         tables, domain, source_var = [], None, None
         if self.primary_data is not None:
             tables = [self.primary_data] + list(self.more_data.values())
             domain = self.primary_data.domain
         elif self.more_data:
             tables = self.more_data.values()
-            if self.merge_type == OWConcatenate.MergeUnion:
-                domain = reduce(domain_union,
-                                (table.domain for table in tables))
-            else:
-                domain = reduce(domain_intersection,
-                                (table.domain for table in tables))
+            domains = [table.domain for table in tables]
+            oper = set.union if self.merge_type == OWConcatenate.MergeUnion \
+                else set.intersection
```

Comment:
Why passing `oper` parameter? It can be set inside `self.merge_domains`.

Reply:
Inertia. :)